### PR TITLE
Add cert manager migration docs

### PR DESCRIPTION
**Pull request by nrfox**
#### What type of PR is this?
Adds migration docs when using cert-manager.

- [ ] Enhancement / New Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Optimization
- [ ] Test
- [x] Documentation Update

#### Which issue(s) this PR fixes:

Fixes #

Related Issue/PR #

### CODE REVIEW

File: docs/ossm/ossm2-migration/cert-manager/README.md [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+      --namespace cert-manager \
+      --wait \
+      --set "app.istio.revisions={basic,ossm3-v1-24-1}"
+  ```
+
+  Depending on whether you will use a `RevisionBased` update strategy or an `InPlace` update strategy, your revision name will vary. If using an `InPlace` strategy, your revision name will match your `Istio` name. If using a `RevisionBased` strategy, revision names use the following format, <istio-name>-v<major_version>-<minor_version>-<patch_version>. For example: `ossm3-v1-24-1`.
+
+- The `app.controller.configmapNamespaceSelector` field needs to be either unset _before_ the migration begins or updated _after_ you have completed your migration.
+
+  If you have set the `app.controller.configmapNamespaceSelector` field on your istio-csr deployment to `maistra.io/member-of`, you will need to update this accordingly. If you haven't set this field, you can keep it unset.
```
Comment (FilipB) [inline docs/ossm/ossm2-migration/cert-manager/README.md:255]: Is it possible to have multiple labels defined in `app.controller.configmapNamespaceSelector`? It would be nice to include also some example when you need to keep `maistra.io/member-of` so 2.x is still working while 3.0 namespace would use a different label.

**Review by FilipB — CHANGES_REQUESTED** [review]

File: docs/ossm/ossm2-migration/cert-manager/README.md [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+    secretName: istio-ca
+```
+
+1. Helm install istio-csr
+
+```console
+helm repo add jetstack https://charts.jetstack.io --force-update
+helm upgrade cert-manager-istio-csr jetstack/cert-manager-istio-csr \
+    --install \
+    --namespace cert-manager \
```
[the remark this answers is not in the document]
Comment (nrfox) [reply docs/ossm/ossm2-migration/cert-manager/README.md:86]: iirc if you use `istio-system` you have to `--set` more options when installing istio-csr. iirc when I originally wrote the cert-manager docs I was using the [upstream docs](https://cert-manager.io/docs/usage/istio-csr/installation/#5-installing-istio-csr) which has you install istio-csr in the `cert-manager` namespace so that's why it's different.

File: docs/ossm/ossm2-migration/cert-manager/README.md [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+  ```console
+  helm upgrade cert-manager-istio-csr jetstack/cert-manager-istio-csr \
+      --install \
+      --reuse-values \
+      --namespace cert-manager \
+      --wait \
+      --set "app.istio.revisions={basic,ossm3-v1-24-1}"
+  ```
+
+  Depending on whether you will use a `RevisionBased` update strategy or an `InPlace` update strategy, your revision name will vary. If using an `InPlace` strategy, your revision name will match your `Istio` name. If using a `RevisionBased` strategy, revision names use the following format, <istio-name>-v<major_version>-<minor_version>-<patch_version>. For example: `ossm3-v1-24-1`.
```
[the remark this answers is not in the document]
Comment (nrfox) [reply docs/ossm/ossm2-migration/cert-manager/README.md:251]: Ah you're right. Must be some markdown processing error.

File: docs/ossm/ossm2-migration/cert-manager/README.md (hunk 1 above)
Comment (nrfox) [reply docs/ossm/ossm2-migration/cert-manager/README.md:255]: >It would be nice to include also some example when you need to keep maistra.io/member-of so 2.x is still working while 3.0 namespace would use a different label.

This is where you need to update the selector **only after** completing the migration and have also added the new injection label to each namespace.

File: docs/ossm/ossm2-migration/cert-manager/README.md [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+
+  To unset this field:
+
+  ```console
+  helm upgrade cert-manager-istio-csr jetstack/cert-manager-istio-csr \
+      --install \
+      --reuse-values \
+      --namespace cert-manager \
+      --wait \
+      --set "app.controller.configmapNamespaceSelector=null"
```
Comment (FilipB) [inline docs/ossm/ossm2-migration/cert-manager/README.md:267]: This was not working for me, the value was unchanged after this. I used `--set "app.controller.configmapNamespaceSelector="`.

**Comment by FilipB** [comment]
Verified that:
-  `istio-ca-root-cert` is unchanged when updating configmapNamespaceSelector
-  `istio-ca-root-cert` is unchanged when re-created after removal
- communication between sleep 3.0 (after upgrade) <> httpbin 2.6 still works

File: docs/ossm/ossm2-migration/cert-manager/README.md (hunk 4 above)
Comment (nrfox) [reply docs/ossm/ossm2-migration/cert-manager/README.md:267]: I must've copied an array example that used `null`

File: docs/ossm/ossm2-migration/cert-manager/README.md (hunk 1 above)
Comment (nrfox) [reply docs/ossm/ossm2-migration/cert-manager/README.md:255]: I'll reword it to make it more clear that you don't always need the injection label.

File: docs/ossm/ossm2-migration/cert-manager/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (nrfox) [reply docs/ossm/ossm2-migration/cert-manager/README.md:86]: The 3.0 documentation uses `cert-manager`: https://docs.redhat.com/en/documentation/red_hat_openshift_service_mesh/3.0.0tp1/html/installing/ossm-cert-manager-assembly#inplace-istio-csr-installation_ossm-cert-manager-assembly so it's a mixed bag. I can update the namespace here to be `istio-system` we'll also need to update the existing 3.0 docs if we want everything to match.

File: docs/ossm/ossm2-migration/cert-manager/README.md [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+spec:
+  ca:
+    secretName: istio-ca
+```
+
+1. Helm install istio-csr
+
+```console
+helm repo add jetstack https://charts.jetstack.io --force-update
+helm upgrade cert-manager-istio-csr jetstack/cert-manager-istio-csr \
```
Comment (FilipB) [inline docs/ossm/ossm2-migration/cert-manager/README.md:84]: Sorry for complication but since this is 2.6 -> 3.0 migration guide, why not just following the 2.6 OSSM [documentation](https://docs.openshift.com/container-platform/4.16/service_mesh/v2x/ossm-security.html#ossm-cert-manager-installation_ossm-security) as a starting point? That way we will have the same starting point as our customers.

File: docs/ossm/ossm2-migration/cert-manager/README.md [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+  gateways:
+    enabled: false
+    openshiftRoute:
+      enabled: false
+  profiles:
+    - default
+  security:
+    certificateAuthority:
+      cert-manager:
+        address: cert-manager-istio-csr.cert-manager.svc:443
```
Comment (FilipB) [inline docs/ossm/ossm2-migration/cert-manager/README.md:123]: I think this should be `cert-manager-istio-csr.istio-system.svc:443`

File: docs/ossm/ossm2-migration/cert-manager/README.md [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+  version: v2.6
+```
+
+1. Update istio-csr with future Istio revision.
+
+```console
+helm upgrade cert-manager-istio-csr jetstack/cert-manager-istio-csr \
+    --install \
+    --reuse-values \
+    --namespace cert-manager \
```
Comment (FilipB) [inline docs/ossm/ossm2-migration/cert-manager/README.md:141]:
```suggestion
    --namespace istio-system \
```

File: docs/ossm/ossm2-migration/cert-manager/README.md [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+metadata:
+  generation: 3
+  name: ossm3
+spec:
+  namespace: istio-system
+  updateStrategy:
+    type: RevisionBased
+  values:
+    global:
+      caAddress: cert-manager-istio-csr.cert-manager.svc:443
```
Comment (FilipB) [inline docs/ossm/ossm2-migration/cert-manager/README.md:160]:
```suggestion
      caAddress: cert-manager-istio-csr.istio-system.svc:443
```

File: docs/ossm/ossm2-migration/cert-manager/README.md [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+kind: ServiceMeshControlPlane
+metadata:
+  name: basic
+  namespace: istio-system
+spec:
+  ...
+  security:
+    certificateAuthority:
+      cert-manager:
+        address: cert-manager-istio-csr.cert-manager.svc:443
```
Comment (FilipB) [inline docs/ossm/ossm2-migration/cert-manager/README.md:234]:
```suggestion
        address: cert-manager-istio-csr.istio-system.svc:443
```

File: docs/ossm/ossm2-migration/cert-manager/README.md [hunk 10]
```diff
@@ -0,0 +1,310 @@
...
+  apiVersion: sailoperator.io/v1alpha1
+  kind: Istio
+  metadata:
+    name: ossm3
+  spec:
+    ...
+    namespace: istio-system
+    values:
+      global:
+        caAddress: cert-manager-istio-csr.cert-manager.svc:443
```
Comment (FilipB) [inline docs/ossm/ossm2-migration/cert-manager/README.md:304]:
```suggestion
        caAddress: cert-manager-istio-csr.istio-system.svc:443
```

File: docs/ossm/ossm2-migration/cert-manager/README.md (hunk 5 above)
Comment (nrfox) [reply docs/ossm/ossm2-migration/cert-manager/README.md:84]: That makes sense. I was following the [3.0 OSSM docs for cert-manager](https://docs.redhat.com/en/documentation/red_hat_openshift_service_mesh/3.0.0tp1/html/installing/ossm-cert-manager-assembly#ossm-installing-cert-manager_ossm-cert-manager-assembly).

**Review by FilipB — APPROVED** [review]

File: docs/ossm/ossm2-migration/cert-manager/README.md [hunk 11]
```diff
@@ -0,0 +1,336 @@
+# OpenShift Service Mesh 2.6 --> 3.0 Migration with Cert-Manager
+
+When migrating from OpenShift Service Mesh 2.6 --> 3.0 while using Cert-Manager you can largely follow the [ClusterWide](TODO) or [MultiTenant](../multi-tenancy/README.md) migration guides with a few extra steps detailed here.
```
Comment (cam-garrison) [inline docs/ossm/ossm2-migration/cert-manager/README.md:3]:
```suggestion
When migrating from OpenShift Service Mesh 2.6 --> 3.0 while using Cert-Manager you can largely follow the [ClusterWide](TODO) or [MultiTenant](../multi-tenancy/README.md) migration guides. This document details a few necessary additional steps to follow before creating your `Istio` resource to ensure your cert manager configuration works with 3.0. 
```

**Review by cam-garrison — APPROVED** [review]
Added a suggestion to make the order of when to do these steps more clear at the beginning of the doc. Otherwise it LGTM

**Comment by nrfox** [comment]
@FilipB CI keeps failing on different stages. Might need blanket overrides.

**Comment by FilipB** [comment]
Doc only PR, overriding the tests.

**Comment by FilipB** [comment]
/override ci/prow/lint ci/prow/e2e-ocp-4-17 ci/prow/e2e-ocp-4-17-arm ci/prow/e2e-ocp-4-18 ci/prow/gencheck ci/prow/integration ci/prow/scorecard ci/prow/unit
